cypress/integration/projectSettings/project_settings.ts

```diff
@@ -253,13 +271,14 @@ describe("Project Settings when defaulting to repo", () => {
     cy.preserveCookies();
   });
 
-  it("Loads with the save button disabled initially", () => {
+  it("Should not have the save button enabled on load", () => {
     cy.dataCy("save-settings-button").should("be.disabled");
   });
 
   it("Preserves edits to the form when navigating between settings tabs and does not show a warning modal", () => {
     cy.dataCy("spawn-host-input").should("have.value", "/path");
     cy.dataCy("spawn-host-input").type("/test");
+    cy.wait(HAS_CHANGES_TIMEOUT_MS); // eslint-disable-line cypress/no-unnecessary-waiting
```

Comment:
I experimented with using `cy.clock()` and `cy.tick()` but that had unexpected consequences on debounce in later tests, even after restoring the clock. Since we only need to wait in this one place I think it's no big deal.

Reply:
Since `setHasChanges` is debounced, we need to make sure it finishes running and sets `hasChanges` to true so that the modal appears on line 283.

Reply:
Oh sorry I got confused above—without waiting for debounce to complete, the "Shows a navigation warning modal when clicking on a header link" test will fail because the test will navigate to a new page via header link

Reply:
Would an assertion on the `Save Changes on Page` button state work instead?     

`cy.dataCy("save-settings-button").should("not.be.disabled");`

I'm hesitant on arbitrary waits since they tend to be flaky and unnecessarily slow down the test suite. We removed several of them in the past.

Reply:
Yeah I understand what you're saying. But I think this is a unique case because we are testing that state is preserved when navigating to a new page (via `cy.dataCy("navitem-access").click()`). If there is no wait, then `setHasChanges` will not fire before loading the Access page and then we aren't able to effectively test that state is preserved between pages. Def agree about avoiding waits in general but I think there are a few uncommon things happening here that necessitate it.

Reply:
Sorry i should have clarified what i meant by the `Save Changes on Page` assertion. If you instead asserted that the `Save Changes on Page` button became un-disabled  it would accomplish the same affect as waiting for the debounce since its state only changes once the validation runs. It would have offered a similar outcome to waiting to the debounce without relying on a set interval.